File: src/main/java/turniplabs/industry/block/BlockLeavesRubber.java
package turniplabs.industry.block;

import net.minecraft.src.*;
import turniplabs.industry.ModIndustry2;

import java.util.Random;

public class BlockLeavesRubber extends BlockLeavesBase {

    public BlockLeavesRubber(int i, Material material, boolean flag) {
        super(i, material, flag);
        setLightOpacity(1);
    }

    @Override
    public int getRenderColor(int i) {
        return ColorProperties.fRGB2iRGB(ColorProperties.pine.inventoryR, ColorProperties.pine.inventoryG, ColorProperties.pine.inventoryB);
    }

    @Override
    public int colorMultiplier(World world, IBlockAccess iblockaccess, int i, int j, int k) {
        int baseFoliageColor = ColorProperties.fRGB2iRGB(ColorProperties.pine.inventoryR, ColorProperties.pine.inventoryG, ColorProperties.pine.inventoryB);
        if (world != null) {
            double localTemperature = iblockaccess.getWorldChunkManager().getTemperature(i, k);
            double localHumidity = iblockaccess.getWorldChunkManager().getHumidity(i, k);
            Season season = world.getCurrentSeason();
            if (season != null) {
                float progress = world.getSeasonProgress();
                baseFoliageColor = ColorizerFoliage.getSeasonalColor(season, progress, localTemperature, localHumidity, ColorProperties.pine);
            }
        }
        return baseFoliageColor;
    }

    @Override
    public int quantityDropped(int metadata, Random random) {
        return random.nextInt(30) != 0 ? 0 : 1;
    }

    @Override
    public int idDropped(int i, Random random) {
        return ModIndustry2.RUBBER_SAPLING.blockID;
    }
}

File: src/main/kotlin/turniplabs/industry/block/BlockLeavesRubber.kt
package turniplabs.industry.block

import net.minecraft.src.*
import java.util.*

class BlockLeavesRubber(i: Int, material: Material?, flag: Boolean) : BlockLeavesBase(i, material, flag) {

    override fun getRenderColor(i: Int): Int {
        return ColorProperties.fRGB2iRGB(
            ColorProperties.pine.inventoryR,
            ColorProperties.pine.inventoryG,
            ColorProperties.pine.inventoryB
        )
    }

    override fun colorMultiplier(world: World?, iblockaccess: IBlockAccess?, x: Int, y: Int, z: Int): Int {
        var baseFoliageColor = ColorProperties.fRGB2iRGB(
            ColorProperties.pine.inventoryR,
            ColorProperties.pine.inventoryG,
            ColorProperties.pine.inventoryB
        )

        if (world != null) {
            val localTemp: Double? = iblockaccess?.worldChunkManager?.getTemperature(x, z)
            val localHumid: Double? = iblockaccess?.worldChunkManager?.getHumidity(x, z)
            val season: Season = world.currentSeason

            if (season != null) {
                val progress: Float = world.seasonProgress
                baseFoliageColor =
                    ColorizerFoliage.getSeasonalColor(season, progress, localTemp!!, localHumid!!, ColorProperties.pine)
            }
        }
        return baseFoliageColor
    }

    override fun idDropped(i: Int, random: Random?): Int {
        return IndustryBlocks.RUBBER_SAPLING.blockID
    }

    override fun quantityDropped(metadata: Int, random: Random?): Int {
        return if (random?.nextInt(20) == 0) 1 else 0
    }

    override fun harvestBlock(world: World?, entityplayer: EntityPlayer?, x: Int, y: Int, z: Int, meta: Int) {
        if (!world?.isMultiplayerAndNotHost!! &&
            entityplayer?.currentEquippedItem != null &&
            entityplayer.currentEquippedItem?.itemID == Item.toolShears.itemID
        ) {
            entityplayer.addStat(StatList.mineBlockStatArray[blockID], 1)
            world.dropItem(x, y, z, ItemStack(this.blockID, 1, meta))
        } else super.harvestBlock(world, entityplayer, x, y, z, meta)
    }
}